File: workout-tracker/src/app/workouts/models/executed-exercise-dto.ts
import { ExerciseSide } from "../workout/enums/exercise-side";

export class ExecutedExerciseDTO {
  public id: number = 0;
  public createdDateTime: Date = new Date();
  public modifiedDateTime?: Date | null = null;
  public name: string = '';
  public exerciseId: number = 0;
  public resistanceType: number = 0;
  public sequence: number = 0;
  public targetRepCount: number = 0;
  public actualRepCount: number = 0;
  public notes: string = '';
  public resistanceAmount: number = 0;
  public resistanceMakeup: string | null = null;
  public setType: number = 0;
  public duration: number | null = null;
  public formRating: number = 0;
  public rangeOfMotionRating: number = 0;
  public bandsEndToEnd: boolean | null = null;
  public involvesReps: boolean = false;
  public side: ExerciseSide | null = null;
  public usesBilateralResistance: boolean = false;
}

File: API/WorkoutTracker.API/Models/ExecutedExerciseDTO.cs
using System;
using WorkoutTracker.Domain.Exercises;

namespace WorkoutTracker.API.Models
{
    public record ExecutedExerciseDTO(
        int Id, //Not a GUID public ID. This is a child entity and not an aggregate.
        DateTime CreatedDateTime, 
        DateTime? ModifiedDateTime, 
        string Name, 
        Guid ExerciseId,
        ResistanceType ResistanceType,
        byte Sequence,
        byte TargetRepCount,
        byte ActualRepCount,
        string Notes,
        decimal ResistanceAmount,
        string ResistanceMakeup,
        SetType SetType,
        ushort? Duration,
        byte FormRating,
        byte RangeOfMotionRating,
        bool? BandsEndToEnd,
        bool InvolvesReps,
        ExerciseSide? Side,
        bool UsesBilateralResistance);

}
